src/components/Auth/UserFormDialog/controls/CustomAttributes.tsx

```diff
@@ -1,69 +1,68 @@
-import { Button } from '@rmwc/button';
-import { IconButton } from '@rmwc/icon-button';
 import { Typography } from '@rmwc/typography';
 import React from 'react';
-import { useFieldArray } from 'react-hook-form';
 import { FormContextValues } from 'react-hook-form/dist/contextTypes';
 
 import { Field } from '../../../common/Field';
 import { AddAuthUserPayload } from '../../types';
 import styles from './controls.module.scss';
+import { validateSerializedCustomClaims } from './customClaimsValidation';
+
+function validate(attributes: string) {
+  if (attributes === '') {
+    return true;
+  }
+
+  try {
+    // We're reusing server validation which throws.
```

Comment:
nit: we don't really need this comment since it's already clear from the code that it throws. Consider adding a comment to `customClaimsValidation.ts` saying the logic is reused from server (ideally with link to GitHub firebase-tools)

Reply:
Is that part of firebase-tools public yet?

Reply:
Yes.